SourceX/DiabloUI/text_draw.cpp

```diff
@@ -41,15 +41,14 @@ void DrawTTF(const char *text, const SDL_Rect &rectIn, int flags,
 	if (font == NULL || text == NULL || *text == '\0')
 		return;
 	if (*render_cache == NULL) {
-		*render_cache = new TtfSurfaceCache();
 		const auto x_align = XAlignmentFromFlags(flags);
-		(*render_cache)->text = RenderUTF8_Solid_Wrapped(font, text, text_color, rect.w, x_align);
-		ScaleSurfaceToOutput(&(*render_cache)->text);
-		(*render_cache)->shadow = RenderUTF8_Solid_Wrapped(font, text, shadow_color, rect.w, x_align);
-		ScaleSurfaceToOutput(&(*render_cache)->shadow);
+		*render_cache = new TtfSurfaceCache {
+			/*.text=*/ScaleSurfaceToOutput(SDLSurfaceUniquePtr { RenderUTF8_Solid_Wrapped(font, text, text_color, rect.w, x_align) }),
+			/*.shadow=*/ScaleSurfaceToOutput(SDLSurfaceUniquePtr { RenderUTF8_Solid_Wrapped(font, text, shadow_color, rect.w, x_align) }),

```

Comment:
Wait... are these comments some ghetto way of using named arguments? oO

Reply:
yeah im not a fan either, cant the IDE be configured to show them but not add them to the file.

Reply:
In C++20 this is valid syntax to initialize struct fields (without the comment markers): https://abseil.io/tips/172

IDEs can be configured to show these but they will show everything, whereas the programmer can only add these where it makes it clearer.

See https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments

Reply:
It is a shame that this is only available in C++20 indeed.... it can be especially confusing when dealing with raw types since there is zero semantic information on the value itself.

I'm fine with using this (albeit weird) comment convention if we standardize on it. It is better than not having anything.